lib/src/util/childUtil.ts

```diff
@@ -30,3 +30,30 @@ export function only(children: ReactNode, type: React.ComponentType): React.Reac
   });
   return matchingChildren.length ? matchingChildren[0] : null;
 }
+
+/**
+ * @param children to search
+ * @returns the concatenated string contents of thie children
+ */
+export function textContent(children: ReactNode): string {
```

Comment:
I was thinking about this function more and I realized that even doing something like this is limited: it wouldn't work right if actual components were part of the `children`.  What I wish we could find is something like `ReactDOMServer.renderToString()` that only rendered the text nodes and not the element markup.  Though come to think of it even that would have problems if any of the child elements were supposed to be invisible. I wonder if maybe it would be best to just require the caller of `NxTable` to pass in the label strings, or at least to pass them in any time the `children` aren't themselves a single string.

Reply:
Why wouldn't this work right with components instead of children? It will just recurse down and combine all text found into a single string.

Reply:
> Why wouldn't this work right with components instead of children? It will just recurse down and combine all text found into a single string.

I'm referring to essentially the difference between shallow rendering and mounting.  Text that is defined within the implementation of child components will be missed unless you actually render the components.

Reply:
I see what you are getting at, though I suspect that's going to fairly rare for a sortable header. Table headers tend to be pretty static in general.

As an alternative we could define a special component for the sortable headers (`NxSortableHeader` maybe). That would have the advantage of allowing us to define separate requirements such as "simple strings only for children" while still allowing the `NxTableCell` to allow for more complex headers.

Reply:
> As an alternative we could define a special component for the sortable headers (NxSortableHeader maybe). That would have the advantage of allowing us to define separate requirements such as "simple strings only for children" while still allowing the NxTableCell to allow for more complex headers.

That would make it so that, for instance, a sortable header can't have an icon next to the text.  I'd rather stick with what you have now, documenting the caveat, or force users to provide the accessible label separately.